tensorboardX/comet_utils.py

```diff
@@ -0,0 +1,302 @@
+import logging
+import json
+import functools
+from PIL import Image
+from io import BytesIO
+import numpy as np
+from .summary import _clean_tag
+try:
+    import comet_ml
+    comet_installed = True
+except:
+    comet_installed = False
+import torch
+
+
+class CometLogger:
+    def __init__(self, comet=False, **kwargs):
+        global comet_installed
+        self._logging = None
+        self._comet_args = kwargs
+        if comet == False:
+            self._logging = False
+        elif comet == True and comet_installed == False:
+            raise Exception("Comet not installed. Run 'pip install comet-ml'")
+    
+    def _requiresComet(method):
+        @functools.wraps(method)
+        def wrapper(*args, **kwargs):
+            self = args[0]
+            global comet_installed
+            if self._logging is None and comet_installed:
+                self._logging = False
+                try:
+                    comet_ml.init()
```

Comment:
If the `COMET_API_KEY` is passed in, then you can skip comet.init() and then users won't get the message about key not set.

Reply:
@ammaddd Could you make this change to check for api_key in the new config dict, and not call comet.init()?